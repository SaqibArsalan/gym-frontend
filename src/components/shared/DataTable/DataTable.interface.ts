import { TableCellProps } from '@mui/material';
import React from 'react';
import { ITableSearchBarProps } from '../TableSearchBar/TableSearchBar.interface';

// Types

/* eslint-disable-next-line */
type OnRowClick = (value: ITableBodyRow) => void;
/* eslint-disable-next-line */
type OnRowColumnClick = (value: ITableBodyRowColumn) => void;

// Data table interface

export interface IDataTableProps {
	header: ITableHeadProps;
	body: ITableBodyProps;
	customFooter?: any;
	tableStyles?: Object;
	isBorderless?: boolean;
	isStriped?: boolean;
	isStickyHeader?: boolean;
	tableLabel?: string;
	showSearchBar?: boolean;
	searchBarProps?: ITableSearchBarProps;
	containerStyles?: any;
	showLoadMore?: boolean;
	onLoadMore?: () => void;
}

// Table Head Interface

export interface ITableHeadProps {
	columns: ITableHeadColumn[];
}

export interface ITableHeadColumn {
	title?: string;
	defaultCell?: boolean;
	component?: React.Component<any> | React.FunctionComponent<any>;
	componentProps?: any;
	tableCellProps?: TableCellProps;
}

// Table Body Interface

export interface ITableBodyProps {
	rows: ITableBodyRow[];
	bodyStyles?: Object;
}

export interface ITableBodyRow {
	rowIndex?: number;
	onRowClick?: OnRowClick;
	rowStyles?: Object;
	columns: ITableBodyRowColumn[];
}

export interface ITableBodyRowColumn {
	columnIndex?: number;
	onRowColumnClick?: OnRowColumnClick;
	text?: string;
	component?: React.Component<any> | React.FunctionComponent<any>;
	componentProps?: any;
	defaultCell?: boolean;
	tableCellProps?: TableCellProps;
}
