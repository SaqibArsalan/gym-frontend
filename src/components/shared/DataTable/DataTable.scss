@import 'src/assets/scss/Variables.scss';
@import 'src/assets/scss/Extends.scss';

.data-table-paper {
	border-radius: toRem(4);
	.load-more {
		text-transform: uppercase;
		padding-top: toRem(16);
		padding-bottom: toRem(16);
		width: 100%;
		text-align: center;
		font-size: toRem(14);
		color: #69767b;
		cursor: pointer;
		background-color: $lightgrey6;
	}
}
