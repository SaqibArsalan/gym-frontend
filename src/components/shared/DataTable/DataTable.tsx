import React from 'react';

import {
	TableContainer,
	Paper,
	Table,
	TableFooter,
	TableRow,
} from '@mui/material';

import { IDataTableProps } from './DataTable.interface';
import DataTableHead from './DataTableHead/DataTableHead';
import DataTableBody from './DataTableBody/DataTableBody';
import CardLabel from '../CardLabel/CardLabel';

import './DataTable.scss';
import TableSearchBar from '../TableSearchBar/TableSearchBar';

/* eslint-disable */
const DataTable: React.FunctionComponent<IDataTableProps> = (
	props: IDataTableProps
) => {
	const {
		tableStyles,
		isStickyHeader,
		header,
		body,
		tableLabel,
		showSearchBar,
		searchBarProps,
		containerStyles,
		showLoadMore,
		onLoadMore,
	} = props;

	// sx: {
	// 	width: '50%',
	// },

	return (
		<Paper className='data-table-paper' variant='outlined'>
			<TableContainer sx={containerStyles || {}}>
				{tableLabel ? <CardLabel label={tableLabel} /> : null}
				{showSearchBar ? <TableSearchBar {...(searchBarProps || {})} /> : null}
				<Table sx={tableStyles || {}} stickyHeader={isStickyHeader}>
					<DataTableHead {...header} />
					<DataTableBody {...body} />
				</Table>
				{showLoadMore ? (
					<div className='load-more' onClick={onLoadMore}>
						Load More
					</div>
				) : null}
			</TableContainer>
		</Paper>
	);
};

export default DataTable;
