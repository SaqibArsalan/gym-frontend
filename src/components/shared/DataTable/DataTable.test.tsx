import React from 'react';
import '@testing-library/react/dont-cleanup-after-each';
import { cleanup, render, RenderResult } from '@testing-library/react';
import '@testing-library/jest-dom';
import DataTable from './DataTable';

let renderWrapper: RenderResult;

// const mockHookDispatch = jest.fn();

// jest.mock('react-redux', () => ({
// 	...jest.requireActual('react-redux'),
// 	useDispatch: () => mockHookDispatch,
// }));

describe('MultiSelectComponent tests', () => {
	beforeAll(() => {
		renderWrapper = render(
			<DataTable
				header={{ columns: [{ title: 'test' }] }}
				body={{ rows: [{ columns: [{ text: '' }] }] }}
			/>
		);
	});
	afterAll(() => {
		cleanup();
	});
	test('default renders successfully', () => {
		expect(renderWrapper.container).not.toBeNull();
		expect(renderWrapper.container).toMatchSnapshot();
	});
});
