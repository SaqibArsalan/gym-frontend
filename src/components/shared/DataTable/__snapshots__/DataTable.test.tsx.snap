// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultiSelectComponent tests default renders successfully 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded data-table-paper css-172kgz0-MuiPaper-root"
  >
    <div
      class="MuiTableContainer-root css-rorn0c-MuiTableContainer-root"
    >
      <table
        class="MuiTable-root css-rqglhn-MuiTable-root"
      >
        <thead
          class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-ur18s6-MuiTableCell-root"
              scope="col"
            >
              test
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root data-table-row css-11toj2m-MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1basyhb-MuiTableCell-root"
            />
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
